### PR TITLE
test(ingest/unity): Add Unity Catalog memory performance testing

### PR DESCRIPTION
**Pull request by asikowitz**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/performance/databricks/test_unity.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import logging
+import os
+from typing import Iterable, Tuple
+from unittest.mock import patch
+
+import humanfriendly
+import psutil
+from performance.databricks.unity_proxy_mock import UnityCatalogApiProxyMock
```
Comment (hsheth2) [inline metadata-ingestion/tests/performance/databricks/test_unity.py:8]: these imports should be `tests.performance...`

File: metadata-ingestion/tests/performance/databricks/test_unity.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        print(f"Workunits Generated: {num_workunits}")
+        print(f"Seconds Elapsed: {timer.elapsed_seconds():.2f} seconds")
+
+    print(
+        f"Peak Memory Used: {humanfriendly.format_size(peak_memory_usage - pre_mem_usage)}"
+    )
+    print(source.report.aspects)
+
+
+def workunit_sink(workunits: Iterable[MetadataWorkUnit]) -> Tuple[int, int]:
```
Comment (hsheth2) [inline metadata-ingestion/tests/performance/databricks/test_unity.py:68]: this looks like copied code - any chance we could refactor some of the common logic out?

**Review by hsheth2 — APPROVED** [review]
few comments but overall looks reasonable

File: metadata-ingestion/tests/performance/databricks/test_unity.py (hunk 2 above)
Comment (asikowitz) [reply metadata-ingestion/tests/performance/databricks/test_unity.py:68]: Nice catch

**Comment by asikowitz** [comment]
Merging through unrelated smoke test flake

**Pull request merged**